libcrux-ml-kem/src/mlkem768.rs

```diff
@@ -56,11 +56,13 @@ macro_rules! instantiate {
             ///
             /// Returns `true` if valid, and `false` otherwise.
             pub fn validate_public_key(public_key: &MlKem768PublicKey) -> bool {
+                #[allow(unused_unsafe)]
```

Comment:
We have all these unsafes everywhere now. Shall we put safe wrappers into `avx2.rs` instead? Then the `unsafe` is contained to that file.